### PR TITLE
Don't break blocks with swords in creative

### PR DESCRIPTION
**Pull request by Commandcracker**
## Description
You can't break blocks in creative while holding those items.

## Testing
tested with those items.

## Checklist
Things need to be done before this Pull Request can be merged.

- [x] Code is well-formatted and adheres to project style guidelines: `cargo fmt`
- [x] Code does not produce any clippy warnings: `cargo clippy`
- [x] All unit tests pass: `cargo test`
- [ ] I added new unit tests, so other people don't accidentally break my code by changing other parts of the codebase. [How?](https://doc.rust-lang.org/book/ch11-01-writing-tests.html)

### CODE REVIEW

File: pumpkin/src/client/player_packet.rs [hunk 1]
```diff
@@ -84,6 +85,22 @@ impl PumpkinError for BlockPlacingError {
...
 }
 
+static CREATIVE_DONT_BREAK_WITH: LazyLock<HashSet<u16>> = LazyLock::new(|| {
+    [
+        get_item("wooden_sword").unwrap().id,
+        get_item("stone_sword").unwrap().id,
+        get_item("iron_sword").unwrap().id,
+        get_item("golden_sword").unwrap().id,
+        get_item("diamond_sword").unwrap().id,
+        get_item("netherite_sword").unwrap().id,
```
Comment (DataM0del) [inline pumpkin/src/client/player_packet.rs:95]: Why not just check if it's a sword instead of having a hardcoded list of items?

Comment (DataM0del) [inline pumpkin/src/client/player_packet.rs:95]: Why not just check if it's a sword instead of having a hardcoded list of items?

Comment (Commandcracker) [reply pumpkin/src/client/player_packet.rs:95]: is_sword is also hardcoded, and it is missing the trident and mace

**Comment by Snowiiii** [comment]
This is something we want to have abstractions to work with. I would consider an System simliar to the current Interactive blocks System. In the future we will have to add more logic to items and for example a method like Mojang uses `canMine` would be great

**Comment by OfficialKris** [comment]
#350 would also benefit from an "item manager." This seems to be pretty important, especially with the rise in added functionality to individual items. I can start work on it once I finish #388 or someone else can give it a shot. Currently, the handle functions have too much going on. Those function should only find out which item is being used and then run the corresponding category, modifier, and specific item code (pickaxes, enchantments, spawn eggs).

File: pumpkin/src/client/player_packet.rs (hunk 1 above)
Comment (DataM0del) [reply pumpkin/src/client/player_packet.rs:95]: > is_sword is also hardcoded, and it is missing the trident and mace

:facepalm:
something filtering over all items could filter using `contains` and checking if the item name contains `sword`
```rs
itemName.contains("sword")
```
Also, doesn't Minecraft have metadata for which category or what type of item this is (e.g. swords)?

**Pull request closed without merging**